src/PlaywrightEnvironment.ts

```diff
@@ -159,10 +181,22 @@ export const getPlaywrightEnv = (basicEnv = 'node') => {
     }
 
     async teardown(): Promise<void> {
-      const { page, browser } = this.global
+      const { browser, context, page } = this.global
+      const { collectCoverage } = this._jestPlaywrightConfig
       if (page) {
         page.removeListener('pageerror', handleError)
       }
+      if (collectCoverage) {
+        await Promise.all(
+          (context as BrowserContext).pages().map((p) =>
+            p.close({
+              runBeforeUnload: true,
+            }),
+          ),
+        )
+        // wait until coverage data was sent successfully to the exposed function
+        await new Promise((resolve) => setTimeout(resolve, 10))
```

Comment:
I found no other way of doing that. By using a sleep of 10ms it should be enough to let the event loop handle to execute the beforeunload handling and its corresponding disk write.